helm-charts/common/ui/templates/configmap.yaml

```diff
@@ -8,14 +8,24 @@ metadata:
   labels:
     {{- include "ui.labels" . | nindent 4 }}
 data:
-  APP_BACKEND_SERVICE_ENDPOINT: {{ .Values.BACKEND_SERVICE_ENDPOINT | quote }}
-  APP_DATA_PREP_SERVICE_URL: {{ .Values.DATAPREP_SERVICE_ENDPOINT | quote }}
+  {{- if contains "codegen-react-ui" .Values.image.repository }}
+  VITE_CODE_GEN_URL: {{ .Values.BACKEND_SERVICE_ENDPOINT | quote }}
+  {{- else if contains "codegen-ui" .Values.image.repository }}
+  BASIC_URL: {{ .Values.BACKEND_SERVICE_ENDPOINT | quote }}
+  {{- else if contains "codetrans-ui" .Values.image.repository }}
+  BASE_URL: {{ .Values.BACKEND_SERVICE_ENDPOINT | quote }}
+  {{- else if contains "docsum-ui" .Values.image.repository }}
+  DOC_BASE_URL: {{ .Values.BACKEND_SERVICE_ENDPOINT | quote }}
+  {{- else if contains "docsum-react-ui" .Values.image.repository }}
+  VITE_DOC_SUM_URL: {{ .Values.BACKEND_SERVICE_ENDPOINT | quote }}
+  {{- else if contains "chatqna-ui" .Values.image.repository }}
   CHAT_BASE_URL: {{ .Values.BACKEND_SERVICE_ENDPOINT | quote }}
   UPLOAD_FILE_BASE_URL: {{ .Values.DATAPREP_SERVICE_ENDPOINT | quote }}
   GET_FILE: {{ .Values.DATAPREP_GET_FILE_ENDPOINT | quote }}
   DELETE_FILE: {{ .Values.DATAPREP_DELETE_FILE_ENDPOINT | quote }}
-  BASE_URL: {{ .Values.BACKEND_SERVICE_ENDPOINT | quote }}
-  DOC_BASE_URL: {{ .Values.BACKEND_SERVICE_ENDPOINT | quote }}
-  BASIC_URL: {{ .Values.BACKEND_SERVICE_ENDPOINT | quote }}
-  VITE_CODE_GEN_URL: {{ .Values.BACKEND_SERVICE_ENDPOINT | quote }}
-  VITE_DOC_SUM_URL: {{ .Values.BACKEND_SERVICE_ENDPOINT | quote }}
+  {{- else }}
```

Comment:
maybe we should have same check logic for chatqna conversational ui related config too? i.e. 'else if contains'

and use helm `fail` to fail other non-supported images?